frontend/src/components/LocationSearch/CurrentLocationButton.tsx

```diff
@@ -0,0 +1,39 @@
+import { IonButton } from "@ionic/react";
+import "leaflet/dist/leaflet.css";
+import { Location } from "../../api";
+import { useCurrentLocation } from "../../hooks/useCurrentLocation";
+import { useLocationFinderApi } from "../../services/apiClients/ApiClientsContext";
+
+export interface CurrentLocationButtonProps {
+  onButtonClicked: (location: Location) => void;
+}
+
+export function CurrnetLocationButton(props: CurrentLocationButtonProps): JSX.Element {
+
+  const currentLocation = useCurrentLocation();
+  const api = useLocationFinderApi();
+
+  const setCurrentLocationAsInputValue = async (): Promise<void> => {
+
+    if (currentLocation.state !== "located") {
+      alert("No current location detected!");
+      return;
+    }
+
+    const locationWithAssinedStops = await api.locationFinderControllerFindLocationsAtCoordinates({
+      latitude: currentLocation.location.coords.latitude,
+      longitude: currentLocation.location.coords.longitude
+    });
+
+    props.onButtonClicked({
+      id: locationWithAssinedStops.id,
+      name: locationWithAssinedStops.name,
+      details: locationWithAssinedStops.details,
+      type: locationWithAssinedStops.type
+    });
```

Comment:
Hier fehlta auch noch komplett die Fehlerbehandlung wenn 

```
const locationWithAssinedStops = await api.locationFinderControllerFindLocationsAtCoordinates({
```

fehlschlägt.

Reply:
Exception handling is implemented.